src/Abp/WebHooks/BackgroundWorker/WebHookSenderJob.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using Abp.BackgroundJobs;
+using Abp.Dependency;
+
+namespace Abp.Webhooks.BackgroundWorker
+{
+    public class WebhookSenderJob : BackgroundJob<WebhookSenderArgs>, ITransientDependency
+    {
+        private readonly IIocResolver _iocResolver;
+        private readonly IBackgroundJobManager _backgroundJobManager;
+        private readonly IWebhooksConfiguration _webhooksConfiguration;
+
+        public WebhookSenderJob(
+            IIocResolver iocResolver,
+            IBackgroundJobManager backgroundJobManager,
+            IWebhooksConfiguration webhooksConfiguration)
+        {
+            _iocResolver = iocResolver;
+            _backgroundJobManager = backgroundJobManager;
+            _webhooksConfiguration = webhooksConfiguration;
+        }
+
+        public override void Execute(WebhookSenderArgs args)
+        {
+            if (args.WebhookId == default)
+            {
+                throw new ArgumentNullException(nameof(args.WebhookId));
+            }
+
+            if (args.WebhookSubscriptionId == default)
+            {
+                throw new ArgumentNullException(nameof(args.WebhookSubscriptionId));
+            }
+
+            using (var webhookSender = _iocResolver.ResolveAsDisposable<IWebhookSender>())
+            {
+                if (webhookSender.Object.TrySendWebhook(args))
+                {
+                    return;
+                }
+
+                using (var workItemStore = _iocResolver.ResolveAsDisposable<IWebhookSendAttemptStore>())
+                {
+                    var sendAttemptCount = workItemStore.Object.GetSendAttemptCount(args.TenantId, args.WebhookId, args.WebhookSubscriptionId);
+
+                    if (sendAttemptCount < _webhooksConfiguration.MaxSendAttemptCount)
+                    {
+                        //try send again
+                        _backgroundJobManager.Enqueue<WebhookSenderJob, WebhookSenderArgs>(args);
+                    }
```

Comment:
If an http request fails, the next request may still fail. In my impression, the http callback will wait for a certain time to send the second request.

eg:
15s/30s/3m/10m/20m/30m/60m/3h/6h/.


In fact, BackgroundJobManager has a built-in retry function.If an exception occurs in a job, it will retry after a delay.
60\120\240\480\960\1920..

Reply:
Thanks @maliming 
Let me check that and change it.